linera-base/src/data_types.rs

```diff
@@ -952,125 +957,60 @@ impl CompressedBytecode {
     }
 }
 
-/// Internal bytes of a blob.
-#[derive(Clone, Hash, Serialize, Deserialize)]
-#[cfg_attr(with_testing, derive(Eq, PartialEq))]
-#[repr(transparent)]
-pub struct BlobBytes<'a>(#[serde(with = "serde_bytes")] pub &'a [u8]);
-
-impl<'a> BcsHashable<'a> for BlobBytes<'a> {}
+impl<'a> BcsHashable<'a> for BlobContent {}
 
 /// A blob of binary data.
-#[derive(Hash, Clone, Debug, Serialize, Deserialize, WitType, WitStore)]
+#[derive(Hash, Clone, Debug, Serialize, Deserialize)]
 #[cfg_attr(with_testing, derive(Eq, PartialEq))]
-pub enum BlobContent {
-    /// A generic data blob.
-    Data(
-        #[serde(with = "serde_bytes")]
-        #[debug(skip)]
-        Vec<u8>,
-    ),
-    /// A blob containing contract bytecode.
-    ContractBytecode(#[debug(skip)] CompressedBytecode),
-    /// A blob containing service bytecode.
-    ServiceBytecode(#[debug(skip)] CompressedBytecode),
+pub struct BlobContent {
+    /// The type of data represented by the bytes.
+    blob_type: BlobType,
+    /// The binary data.
+    #[serde(with = "serde_bytes")]
+    bytes: Box<[u8]>,
```

Comment:
Are these compressed or de-* bytes now? This information is now missing. Can you add that as a comment?

Reply:
We currently only compress bytecode, but not other data. I'll add a comment to the `BlobType` variant. This type here is unaware of what the bytes mean or are used for.